File: game_of_life/GameOfLifeJL/test/GridTests.jl
using Test

include("../src/Grid.jl")

@testset "Grid" begin
    @testset "create grid" begin
        grid = create_grid(3,4,false)
        @test size(grid.data) == (3,4)
    end

    @testset "get_cell" begin
        grid = create_grid(3,4,false)
        @test get_cell(grid, 1, 1) == false
        @test get_cell(grid, 2, 3) == false
    end

    @testset "set_cell" begin
        grid = create_grid(3,4,false)
        set_cell!(grid, 1, 1, true)
        @test get_cell(grid, 1, 1) == true
        set_cell!(grid, 2, 3, true)
        @test get_cell(grid, 2, 3) == true
    end

    @testset "map_grid" begin
        grid = create_grid(3,4,false)
        map_grid!(x -> !x, grid)
        # @test get_cell(grid, 1, 1) == true
        # @test get_cell(grid, 2, 3) == false
        @test all( x-> x == true, grid.data)
    end
end

File: game_of_life/GameOfLifeFS/tests/GridTests.fs
// tests/GridTests.fs
namespace GameOfLife.Tests

open NUnit.Framework
open GameOfLife.Grid

[<TestFixture>]
type GridTests() =

    [<Test>]
    member this.``Create grid with correct dimensions``() =
        let grid = createGrid 3 4 false
        Assert.AreEqual(grid.Rows, 2)
        Assert.AreEqual(grid.Cols, 4)

    [<Test>]
    member this.``Get cell value``() =
        let grid = createGrid 3 4 false
        Assert.AreEqual(getCell grid (1, 2), false)

    [<Test>]
    member this.``Set cell value``() =
        let grid = createGrid 3 4 false
        let newGrid = setCell grid (1, 2) true
        Assert.AreEqual(getCell newGrid (1, 2), true)

    [<Test>]
    member this.``Map function over grid``() =
        let grid = createGrid 3 4 false
        let newGrid = mapGrid (fun x -> not x) grid
        Assert.AreEqual(getCell newGrid (1, 2), true)